services/provider/main.go

```diff
@@ -13,6 +17,62 @@ var (
 
 func main() {
 	flag.Parse()
-	var opts []grpc.ServerOption
-	server.Start(*port, opts)
+
+	klog.Info("Starting Provider API server")
+
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Use `signal.NotifyContext()` instead so that any requests can be canceled if the container main process receives a termination signal. Right now, only the end of execution will terminate this, so it's more or less non-graceful.

Reply:
Thanks for the suggestion. Using `signal.NotifyContext()`.